integration/utils/txTracker.go

```diff
@@ -44,168 +57,178 @@ func NewTxTracker(
+			// if any change in status
+			if tx.lastStatus != result.Status {
+				log = log.With().Dur("duration", time.Since(tx.createdAt)).Logger()
+				switch result.Status {
+				case flowsdk.TransactionStatusFinalized:
+					log.Trace().Msg("tx has been finalized")
+					if tx.onFinalized != nil {
+						go tx.onFinalized(tx.txID, result)
+					}
+					tx.lastStatus = flowsdk.TransactionStatusFinalized
+					tx.stat.TTF = time.Since(tx.createdAt)
+				case flowsdk.TransactionStatusExecuted:
+					log.Trace().Msg("tx has been executed")
+					if tx.onExecuted != nil {
+						go tx.onExecuted(tx.txID, result)
+					}
+					tx.lastStatus = flowsdk.TransactionStatusExecuted
+					tx.stat.TTE = time.Since(tx.createdAt)
+				case flowsdk.TransactionStatusSealed:
+					log.Trace().Msg("tx has been sealed")
+					// sometimes we miss the executed call and just get sealed directly
+					if tx.lastStatus != flowsdk.TransactionStatusExecuted {
 						if tx.onExecuted != nil {
 							go tx.onExecuted(tx.txID, result)
 						}
-						tx.lastStatus = flowsdk.TransactionStatusExecuted
-						tx.stat.TTE = time.Since(tx.createdAt)
-						log.Trace().Int("worker_id", workerID).Str("tx_id", tx.txID.String()).
-							Msgf("tx has been executed in %v seconds", time.Since(tx.createdAt).Seconds())
-					case flowsdk.TransactionStatusSealed:
-						// sometimes we miss the executed call and just get sealed directly
-						if tx.lastStatus != flowsdk.TransactionStatusExecuted {
-							if tx.onExecuted != nil {
-								go tx.onExecuted(tx.txID, result)
-							}
-							// TODO fix me
-							// tx.stat.TTE = time.Since(tx.createdAt)
-						}
-						if tx.onSealed != nil {
-							go tx.onSealed(tx.txID, result)
-						}
+					}
+					if tx.onSealed != nil {
+						go tx.onSealed(tx.txID, result)
+					}
 
-						tx.stat.TTS = time.Since(tx.createdAt)
-						stats.AddTxStats(tx.stat)
-						log.Trace().Int("worker_id", workerID).Str("tx_id", tx.txID.String()).
-							Msgf("tx has been sealed in %v seconds", time.Since(tx.createdAt).Seconds())
-						continue
-					case flowsdk.TransactionStatusUnknown:
-						log.Warn().Int("worker_id", workerID).Str("tx_id", tx.txID.String()).
-							Msgf("got into an unknown status after %v seconds", time.Since(tx.createdAt).Seconds())
-						if tx.onError != nil {
-							go tx.onError(tx.txID, err)
-						}
-						tx.stat.isExpired = true
-						stats.AddTxStats(tx.stat)
-						continue
-					case flowsdk.TransactionStatusExpired:
-						if tx.onExpired != nil {
-							go tx.onExpired(tx.txID)
-						}
-						tx.stat.isExpired = true
-						stats.AddTxStats(tx.stat)
-						log.Warn().Int("worker_id", workerID).Str("tx_id", tx.txID.String()).
-							Msgf("tx has been expired in %v seconds", time.Since(tx.createdAt).Seconds())
-						continue
+					tx.stat.TTS = time.Since(tx.createdAt)
+					txt.stats.AddTxStats(tx.stat)
+					close(tx.wait)
+					continue
+				case flowsdk.TransactionStatusUnknown:
```

Comment:
It's also possible that we start in an "unknown" status, so not sure if it makes sense to throw it into expired right away.

though this is probably impossible for a localnet with only 1 access node

Reply:
Interesting, this will make a difference for benchnet.  I would not complicate this stat machine more though... let me retry the `TransactionStatusUnknown` state until either result or timeout -- this will remove one of the states and make the code cleaner.